openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -39,9 +39,16 @@ def process_request(self, request):
             preferred_language = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
```

Comment:
Outside the scope of this PR:
We should revisit this logic.  Why are browser language settings only honored when the user is unauthenticated?  As we are now doing for timezones, we should always honor browser settings for language as long as it's not overridden in the user's preferences.